### DISCUSSION TITLE
Suggestion for a New Feature to Control Who Can Join the Channel

### DISCUSSION THREAD

**Discussion by muamel1** [BearWare/TeamTalk5#1791] [Ideas]
### Description

I would like to suggest a new option that would allow operators to control who can join the channel. The feature would work as follows:
- When the option is enabled, anyone who wants to join the channel has to request permission first.
- All the operators in the channel receive an alert with the name and other information (such as country and VPN status) of the person who requests to join.
- The operators can then grant or deny the join request based on their judgment.

This feature would help:
- Reduce spam and protect public channels from unwanted intrusions.
- Make it easier for operators to identify and reject spammers based on their information.

I hope that my suggestion will be considered and implemented in future updates.

### Application

- [X] qtTeamTalk
- [X] TeamTalkAndroid
- [X] iTeamTalk
- [ ] TeamTalkServer

**Comment by Ramon-Salazar** [comment]
That's great, this feature will be useful, since it'll be easier to identify spammers and prevent them from joining.

**Comment by amirmahdifard** [comment]
that's why specifying a password for your channel exists, put a password for your channel, and give it to people who you want to join, and if anyone has joined that you don't want them anymore, banning them from the channel and changing the channel password is a way: hope this helps: also, the operator password exists for just in case admins or other operators are not there at that time.

**Comment by muamel1** [comment]
I'm talking about public channels
Imagine this scenario, on your server you have a public channel for all users, someone start spamming this channel: and you can't make this channel private because it's meant to be public.  In this case, my suggestion will be helpful. This is just one of several cases that might be useful if my suggestion is taken into consideration

**Comment by amirmahdifard** [comment]
@Muamalaljanahi bro, if you know what's the meaning of public and what's the meaning of private: public means that everyone should be able to join, because it's public: if you want to controle who joins your channel, then that channel is a private channel, and password feature exists for the private channels: thanks.

**Comment by muamel1** [comment]
@amirmahdifard Are you the developer of TeamTalk?

**Comment by amirmahdifard** [comment]
@Muamalaljanahi of corce i'm not, but if you want to be able to have controle on who joins a public channel, it's not called a public channel anymore: since you haven't found anything else, you are saying if i'm not the developer, i shouldn't have my opinions: but lets see what the developer says about this, but i'm sure that they've made the password feature for such a thing.

**Comment by muamel1** [comment]
> @Muamalaljanahi of corce i'm not, but if you want to be able to have controle on who joins a public channel, it's not called a public channel anymore: since you haven't found anything else, you are saying if i'm not the developer, i shouldn't have my opinions: but lets see what the developer says about this, but i'm sure that they've made the password feature for such a thing.

Well, you can call it whatever you want. But let me take the opportunity to ask you to reduce your comments in this repository if possible. With all due respect, but my email is full of your nonsense comments.

**Comment by amirmahdifard** [comment]
@Muamalaljanahi i guess that the disrespectful person in here is you that you calling my comments noncence: but no worrys, heh if i liked this suggestion that you gave, i would become a good person for you: but it is clear that you've requested that we should be able to controle who can join a channel, and admins can acsept or reject users joining request, so i told you you can put a password for your channel, and you can acsept someones joining request by giving them the password, and you can reject someones joining request by not giving them the password: unless someone else gives a bettar point for this, but currently i don't see any difrence between this suggestion that you've made with the teamtalk channel password features.

**Comment by amirmahdifard** [comment]
@tech10 
hi bro: baced on this person's request and the comments, what do you think about the points of this feature, someone said we can identify spammers with this suggestion in the teamtalk, but i think we can already identify spammers by putting a channel password for the channel, and ban the spammers from the channel, and changing the password: what do you think?

**Comment by tech10** [comment]
@amirmahdifard I think everything here has been sufficiently covered, and all sides of the issue have been examined by others.

**Comment by Nardol** [comment]
@amirmahdifard  OK, now we are watching to Who Wants to Be a Millionaire and use the joker to call a friend?
Calling this comment "nonsense" would definitely be an euphemism and I am really happy to have blocked you on GitHub to stop receiving your comment by e-mail.

**Comment by amirmahdifard** [comment]
@tech10 @Muamalaljanahi ok, so, now i agree with a kind of suggestion in this tikit: we should be able to create a list of some users, that can join our channels with out entering the password, and i think it should be called friends list or something like that, if you also ment to suggest such a thing.

**Comment by muamel1** [comment]
@Nardol Thanks for the idea 😄
